@xen-orchestra/web-core/lib/components/button/UiButton.vue

```diff
@@ -0,0 +1,207 @@
+<template>
+  <button :class="className" :disabled="busy || isDisabled" type="button" class="ui-button">
+    <UiIcon :busy :icon="leftIcon" class="icon" />
+    <slot />
+    <UiIcon :icon="rightIcon" class="icon" />
+  </button>
+</template>
+
+<script lang="ts" setup>
+import UiIcon from '@core/components/icon/UiIcon.vue'
+import { useContext } from '@core/composables/context.composable'
+import { DisabledContext } from '@core/context'
+import type { ButtonLevel, ButtonSize } from '@core/types/button.type'
+import type { Color } from '@core/types/color.type'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { computed } from 'vue'
+
+const props = withDefaults(
+  defineProps<{
+    level?: ButtonLevel
+    busy?: boolean
+    disabled?: boolean
+    leftIcon?: IconDefinition
+    rightIcon?: IconDefinition
+    color?: Color
+    size?: ButtonSize
+  }>(),
+  {
+    disabled: undefined,
+    color: 'info',
+    level: 'primary',
+    size: 'small',
+  }
+)
+
+const isDisabled = useContext(DisabledContext, () => props.disabled)
+
+const fontClasses = {
+  'extra-small': 'typo p3-medium',
+  small: 'typo h6-medium',
+  medium: 'typo h3-semi-bold',
+}
+
+const fontClass = computed(() => fontClasses[props.size])
+
+const className = computed(() => {
+  return [
+    props.color,
+    props.level,
+    props.size,
+    fontClass.value,
+    {
+      busy: props.busy,
+      disabled: isDisabled.value,
+    },
+  ]
+})
```

Comment:
I think we don't need to define the `fontClass` in its own `computed`, since it's only used in another `computed`.

```suggestion
const className = computed(() => {
  return [
    props.color,
    props.level,
    props.size,
    fontClasses[props.size],
    {
      busy: props.busy,
      disabled: isDisabled.value,
    },
  ]
})
```